### ISSUE TITLE
CS8625 code fix fails when declaration to fix is in another file

### ISSUE THREAD

**Issue by jroessel** [dotnet/roslyn#74121]
**Version Used**: 

**Steps to Reproduce**:

1. Have code where code in one file sets a field/property in another file to `null` , e.g.

  **Parent.cs**

  ```csharp
  public class Parent
  {
      public void M(Child c)
      {
          c.next = null;
          c.Next = null;
      }
  }
  ```
  
  **Child.cs**
  
  ```
  public sealed class Child
  {
      internal Child next;
      public Child Next { get => next; set => next = value; }
  }
  ```

2. Place the cursor on either of the `null` literals where CS8625 is raised.
3. Apply the suggested "Declare as nullable" code fix

**Diagnostic Id**: CS8625: Cannot convert null literal to non-nullable reference type.

**Expected Behavior**:

The code fix should append a `?` to the field/property type in the other file.

**Actual Behavior**:

```
System.ArgumentException : The node is not part of the tree.
Parameter name: node
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.CheckNodeInOriginalTree(SyntaxNode node)
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.ReplaceNode(SyntaxNode node,SyntaxNode newNode)
   at async Microsoft.CodeAnalysis.CSharp.CodeFixes.DeclareAsNullable.CSharpDeclareAsNullableCodeFixProvider.FixAllAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeFixes.SyntaxEditorBasedCodeFixProvider.FixAllWithEditorAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetChangedSolutionAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputeOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.ComputePreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeActions.CodeAction.GetPreviewOperationsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.<>c__DisplayClass16_0.<GetPreviewAsync>b__0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

Reproducible in both VS Code and Visual Studio 2022 with .NET 8.0.302.

**Comment by CyrusNajmabadi** [comment]
The code fixer is declared as `internal class CSharpDeclareAsNullableCodeFixProvider : SyntaxEditorBasedCodeFixProvider`.  But this is anot appropriate for this code fixer.  SyntaxEditorBasedCodeFixProvider is correct when the fix will be made in the same lcoation as the diagnostic.  For  CSharpDeclareAsNullableCodeFixProvider  the fix may be in a different location.  This should be made into a normal code fix provider,  And a solution editor should be used for the fix all.
